tests/operators/functionals/test_functionals.py

```diff
@@ -0,0 +1,404 @@
+import inspect
+from collections.abc import Callable, Sequence
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Literal, TypedDict
+
+import pytest
+import torch
+from mrpro.operators import Functional, ProximableFunctional, functionals
+
+from tests import RandomGenerator
+
+
+@dataclass
+class FunctionalTestCase:
```

Comment:
Please add doc-strings here and also to functional_test_cases, test_functional_shape, test_functional_devided_by_n